File: Backend/renderer/src/main/java/io/sparkled/renderer/effect/SplitLineEffectRenderer.java
package io.sparkled.renderer.effect;

import io.sparkled.renderer.context.RenderContext;

public class SplitLineEffectRenderer extends EffectRenderer {

    @Override
    public void render(RenderContext ctx) {
        int start = 0;
        int end = ctx.getChannel().getLedCount() - 1;
        int middle = end / 2;

        LineEffectRenderer lineRenderer = new LineEffectRenderer();
        lineRenderer.renderLine(ctx, start, middle, true);
    }
}

File: renderer/src/main/kotlin/io/sparkled/renderer/effect/SplitLineEffectRenderer.kt
package io.sparkled.renderer.effect

import io.sparkled.renderer.context.RenderContext

class SplitLineEffectRenderer : EffectRenderer() {

    override fun render(ctx: RenderContext) {
        val start = 0
        val end = ctx.channel.ledCount - 1
        val middle = end / 2

        val lineRenderer = LineEffectRenderer()
        lineRenderer.renderLine(ctx, start, middle, true)
    }
}